### PR TITLE
Allow multiple outputs with different root domains for `SDPAOp` in `ComputeAtMap`

### PR DESCRIPTION
**Pull request by Priya2698**
Issue #2751.

### CODE REVIEW

File: tests/cpp/test_sdpa_node.cpp [hunk 1]
```diff
@@ -101,25 +105,40 @@ void checkSdpaFwdMapping(Fusion* fusion, Expr* op) {
       // Idx=3: prodcuer_ids[3] = E/Ev, consumer_idx[3] = Ev
 
       for (auto idx : c10::irange(consumer_ids.size())) {
-        if (idx >= num_device_dim && idx < (2 + num_device_dim)) {
+        if (idx < (2 + num_device_dim)) {
```
Comment (Priya2698) [inline tests/cpp/test_sdpa_node.cpp:108]: Check that DID axis is mapped between `q, k, v` and `output, logsum_exp`.

File: csrc/compute_at_map.cpp [hunk 2]
```diff
@@ -374,6 +374,7 @@ void IterDomainGraph::build(Fusion* fusion) {
       if (first_output_tv == nullptr) {
         first_output_tv = c_tv;
       } else {
+        if (ir_utils::hasUniformSiblings(expr)){
```
Comment (Priya2698) [inline csrc/compute_at_map.cpp:377]: Skip mapping multiple outputs of the expr to each other for `SdpaOps`.

The code at L438 wil map producers and consumers.

**Comment by Priya2698** [comment]
!build

File: csrc/ir/utils.h [hunk 3]
```diff
@@ -650,6 +650,8 @@ std::optional<std::vector<int64_t>> computePermutation(
 
 bool hasTrivialAllocationDomain(const TensorView* tv);
 
+bool hasUniformSiblings(Expr* expr);
```
Comment (jacobhinkle) [inline csrc/ir/utils.h:653]: Comment was lost in the copy-paste
```suggestion
// Returns true if all expr outputs should be mapped unconditionally
bool hasUniformSiblings(Expr* expr);
```

File: csrc/compute_at_map.cpp [hunk 4]
```diff
@@ -374,61 +374,63 @@ void IterDomainGraph::build(Fusion* fusion) {
...
-            c_tv->getLoopDomain(),
-            c2f_root_map);
-
-        // Map the entire replay map between the multiple
-        // consumers
-        auto c2f_disjoint_sets = replay_FasC.getIterDomainEquivalence();
-        for (const auto& disjoint_set : c2f_disjoint_sets.disjointSets()) {
-          if (disjoint_set->empty()) {
-            continue;
+        if (ir_utils::hasUniformSiblings(expr)) {
```
Comment (naoyam) [inline csrc/compute_at_map.cpp:377]: nit: I think you can move this check to the beginning of this loop body, enclosing lines 374 and 375 as well.

**Review by naoyam — APPROVED** [review]
Thanks for the update. LGTM

**Comment by Priya2698** [comment]
!build

**Pull request merged**